File: packages/ssl-pinning/ios/Sources/SSLPinningPlugin/error/SSLPinningErrorMessages.swift
import Foundation

/**
 Canonical error messages shared across platforms.

 These strings must remain byte-identical on iOS and Android.
 */
enum SSLPinningErrorMessages {
    static let urlRequired = "url is required"
    static let invalidUrlMustBeHttps = "Invalid URL. Must be https."
    static let invalidUrl = "Invalid URL."
    static let noFingerprintsProvided = "No fingerprints provided"
    static let noCertsProvided = "No certs provided"
    static let noHostFoundInUrl = "No host found in URL"
    static let invalidFingerprintFormat = "Invalid fingerprint format"
    static let unsupportedHost = "Unsupported host: %@"
    static let pinningFailed = "Pinning failed"
    static let excludedDomain = "Excluded domain"
    static let timeout = "Timeout"
    static let networkError = "Network error"
    static let internalError = "Internal error"
    static let invalidConfig = "Invalid configuration: %@"
    static let certNotFound = "Certificate not found: %@"

    static func unsupportedHost(_ value: String) -> String {
        return String(format: unsupportedHost, value)
    }

    static func invalidConfig(_ value: String) -> String {
        return String(format: invalidConfig, value)
    }

    static func certNotFound(_ value: String) -> String {
        return String(format: certNotFound, value)
    }
}

File: packages/ssl-pinning/android/src/main/java/io/capkit/sslpinning/error/SSLPinningErrorMessages.kt
package io.capkit.sslpinning.error

/**
 * Canonical error messages shared across platforms.
 * These strings must remain byte-identical on iOS and Android.
 */
object SSLPinningErrorMessages {
  const val URL_REQUIRED = "url is required"
  const val INVALID_URL_MUST_BE_HTTPS = "Invalid URL. Must be https."
  const val INVALID_URL = "Invalid URL."
  const val NO_FINGERPRINTS_PROVIDED = "No fingerprints provided"
  const val NO_CERTS_PROVIDED = "No certs provided"
  const val NO_HOST_FOUND_IN_URL = "No host found in URL"
  const val UNSUPPORTED_HOST = "Unsupported host: %s"
  const val PINNING_FAILED = "Pinning failed"
  const val EXCLUDED_DOMAIN = "Excluded domain"
  const val TIMEOUT = "Timeout"
  const val NETWORK_ERROR = "Network error"
  const val INTERNAL_ERROR = "Internal error"

  @JvmStatic
  fun unsupportedHost(value: String): String = String.format(UNSUPPORTED_HOST, value)
}
